python/dataproc_templates/jdbc/README.md

```diff
@@ -0,0 +1,149 @@
+```
+
+## Required JAR files
+
+This template requires the JDBC jar file to be available in the Dataproc cluster.
+User has to download the required jar file and host it inside a GCS Bucket, so that it could be referred during the execution of code.
+
+wget command to download JDBC jar file is as follows :-
+
+* MySQL
+```
+wget http://dev.mysql.com/get/Downloads/Connector-J/mysql-connector-java-5.1.30.tar.gz
+```
+* PostgreSQL
+```
+wget https://jdbc.postgresql.org/download/postgresql-42.2.6.jar
+```
+* Microsoft SQL Server
+```
+wget https://repo1.maven.org/maven2/com/microsoft/sqlserver/mssql-jdbc/6.4.0.jre8/mssql-jdbc-6.4.0.jre8.jar
+```
+
+Once the jar file gets downloaded, please upload the file into a GCS Bucket and export the below variable
+
+```
+export JARS=<gcs-bucket-location-containing-jar-file> 
+```
+
+## JDBC URL syntax
+
+* MySQL
+```
+jdbc:mysql://<hostname>:<port>/<dbname>?user=<username>&password=<password>
+```
+* PostgreSQL
+```
+jdbc:postgresql://<hostname>:<port>/<dbname>?user=<username>&password=<password>
+```
+* Microsoft SQL Server
+```
+jdbc:sqlserver://<hostname>:<port>;databaseName=<dbname>;user=<username>;password=<password>
+```
+
+## Other important properties
+
+* Driver Class
+
+    * MySQL
+    ```
+    jdbctojdbc.input.driver="com.mysql.cj.jdbc.Driver" 
+    ```
+    * PostgreSQL
+    ```
+    jdbctojdbc.input.driver="org.postgresql.Driver"
+    ```
+    * Microsoft SQL Server
+    ```
+    jdbctojdbc.input.driver="com.microsoft.sqlserver.jdbc.SQLServerDriver"
+    ```
+
+* You can either specify the source table name or have SQL query within double quotes. Example,
+
+```
+jdbctojdbc.input.table="employees"
+jdbctojdbc.input.table="(select * from employees where dept_id>10) as employees"
+```
+
+* partitionColumn, lowerBound, upperBound and numPartitions must be used together. If one is specified then all needs to be specified.
+
+* You can specify the target table properties such as partition column using below property. This is useful when target table is not present or when write mode=overwrite and you need the target table to be created as partitioned table.
+
+    * MySQL
+    ```
+    jdbctojdbc.output.create_table.option="PARTITION BY RANGE(id)  (PARTITION p0 VALUES LESS THAN (5),PARTITION p1 VALUES LESS THAN (10),PARTITION p2 VALUES LESS THAN (15),PARTITION p3 VALUES LESS THAN MAXVALUE)"
+    ```
+    * PostgreSQL
+    ```
+    jdbctojdbc.output.create_table.option="PARTITION BY RANGE(id);CREATE TABLE po0 PARTITION OF <table_name> FOR VALUES FROM (MINVALUE) TO (5);CREATE TABLE po1 PARTITION OF <table_name> FOR VALUES FROM (5) TO (10);CREATE TABLE po2 PARTITION OF <table_name> FOR VALUES FROM (10) TO (15);CREATE TABLE po3 PARTITION OF <table_name> FOR VALUES FROM (15) TO (MAXVALUE);"
+    ```
+
+* Additional execution details [refer spark jdbc doc](https://spark.apache.org/docs/latest/sql-data-sources-jdbc.html)
+
+## Example execution: 
```

Comment:
in the example execution please use absolute values.
Refer [here](https://github.com/GoogleCloudPlatform/dataproc-templates/blob/main/java/src/main/java/com/google/cloud/dataproc/templates/jdbc/README.md#2-jdbc-to-gcs)

Reply:
Updated with Absolute values and added for :
- mysql to mysql
- postgresql to postgresql
- SQL server to SQL server
- mysql to postgresql
- mysql to SQL server